sys-libs/ncurses/ncurses6-6.1.recipe

```diff
@@ -0,0 +1,222 @@
+SUMMARY="New curses library"
+DESCRIPTION="The Ncurses (new curses) library is a free software emulation of \
+curses in System V Release 4.0, and more. It uses Terminfo format, supports \
+pads and color and multiple highlights and forms characters and function-key \
+mapping, and has all the other SYSV-curses enhancements over BSD Curses.
+The ncurses code was developed under GNU/Linux. It has been in use for some \
+time with OpenBSD as the system curses library, and on FreeBSD and NetBSD as \
+an external package. It should port easily to any ANSI/POSIX-conforming UNIX. \
+It has even been ported to OS/2 Warp!
+The distribution includes the library and support utilities, including a \
+terminfo compiler tic, a decompiler infocmp, clear, tput, tset, and a \
+termcap conversion tool captoinfo. Full manual pages are provided for the \
+library and tools."
+HOMEPAGE="https://www.gnu.org/software/ncurses/ncurses.html"
+COPYRIGHT="1998-2017 Free Software Foundation, Inc."
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://ftpmirror.gnu.org/ncurses/ncurses-$portVersion.tar.gz
+	https://ftp.gnu.org/gnu/ncurses/ncurses-$portVersion.tar.gz"
+CHECKSUM_SHA256="aa057eeeb4a14d470101eff4597d5833dcef5965331be3528c08d99cebaa0d17"
+SOURCE_DIR="ncurses-$portVersion"
+PATCHES="ncurses-6.1.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64 arm"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+libVersion="$portVersion"
+libVersionCompat="$libVersion compat >= 6.1"
+portVersionCompat="$libVersionCompat"
+
+PROVIDES="
+	ncurses6$secondaryArchSuffix = $portVersionCompat
+	lib:libform$secondaryArchSuffix = $libVersionCompat
+	lib:libformw$secondaryArchSuffix = $libVersionCompat
+	lib:libmenu$secondaryArchSuffix = $libVersionCompat
+	lib:libmenuw$secondaryArchSuffix = $libVersionCompat
+	lib:libncurses$secondaryArchSuffix = $libVersionCompat
+	lib:libncursesw$secondaryArchSuffix = $libVersionCompat
+	lib:libncurses++$secondaryArchSuffix = $libVersionCompat
+	lib:libncurses++w$secondaryArchSuffix = $libVersionCompat
+	lib:libpanel$secondaryArchSuffix = $libVersionCompat
+	lib:libpanelw$secondaryArchSuffix = $libVersionCompat
+	"
+if [ -z "$secondaryArchSuffix" ]; then
+	PROVIDES="$PROVIDES
+		cmd:captoinfo = $portVersion compat >= 5
+		cmd:clear = $portVersion compat >= 5
+		cmd:infocmp = $portVersion compat >= 5
+		cmd:infotocap = $portVersion compat >= 5
+		cmd:reset = $portVersion compat >= 5
+		cmd:tabs = $portVersion compat >= 5
+		cmd:tic = $portVersion compat >= 5
+		cmd:toe = $portVersion compat >= 5
+		cmd:tput = $portVersion compat >= 5
+		cmd:tset = $portVersion compat >= 5
+		"
+fi
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+if [ -n "$secondaryArchSuffix" ]; then
+	REQUIRES="$REQUIRES
+		ncurses6 == $portVersion base
+		"
```

Comment:
```diff
 REQUIRES="
 	haiku$secondaryArchSuffix
 	"
-if [ -n "$secondaryArchSuffix" ]; then
-	REQUIRES="$REQUIRES
-		ncurses6 == $portVersion base
-		"
-fi
```

Reply:
Oops, sorry @extrowerk for having written the above comment.
I though the dependency was there for no reason, but I just saw this comment in `BUILD`:
```
	# For the secondary architecture we don't want the programs nor the
	# terminfo DB. The main package will depend on the primary package which
	# provides all this.
```
So I guess you will need to revert the change.
```diff
 REQUIRES="
 	haiku$secondaryArchSuffix
 	"
+if [ -n "$secondaryArchSuffix" ]; then
+	REQUIRES="$REQUIRES
+		ncurses6 == $portVersion base
+		"
+fi
```

Hmm, that said, maybe that's not such a bad idea. Do we really need to make the base package for the secondary arch depend on that of the primary arch? Well, I don't know. Let's see what the other contributors think.
(Feel free to revert the change when you wish.)